### ISSUE TITLE
Expressions (when) in Lists

### ISSUE THREAD

**Issue by redwolf-albath** [clientIO/joint#1275]
I have been trying to use expressions inside of lists for the inspector, the goal is to have unique list items created based on some selected value inside the list item (or worst case, in the outer scope). 

**Details below:**

I read this post on stack overflow, I'll reference the answer from 'lades' in this issue:
https://stackoverflow.com/questions/37742721/display-field-based-on-another-in-jointjs

My hope is that elements added inside a list can be conditionally different based on a selector inside them.

For example, given some inspector:
```
{
  "inputs": {
    "parameters": {
      "stuff_to_do": {
        "type": "list",
        "label": "General",
        "addButtonLabel": "Add New",
        "removeButtonLabel": "Remove",
        "group": "General",
        "index": 550,
        "item": {
          "type": "object",
          "properties": {
            "condition": {
              "type": "select",
              "defaultValue": "test1",
              "label": "Modify conditional field selection",
              "index": 1,
              "options": [
                {
                  "content": "Test 1",
                  "value": "test1"
                },
                {
                  "content": "Test 2",
                  "value": "test2"
                }
              ]
            },
            "conditional_field": {
              "type": "text",
              "label": "Conditional",
              "index": 2,
              "when": {
                "in": {
                  "parameters/actions/*/action": ["test1"],
                  "dependencies": ["parameters/stuff_to_do/*/condition"]
                }
              }
            }
          }
        }
      }
    }
  }
}
```

The hope is that the selector `"parameters/stuff_to_do/*/condition"` will mean that each list item depends on the value of the `"select"` input in its scope. This is what was proposed in the referenced stack overflow, referencing a potential issue opened that I can't find.

I have also tried:  `"parameters/stuff_to_do/0/condition"` which does work correctly (all list items depend on the value of the first list item's select) but the conditional field only hides/shows when an element is deleted from the list (the list doesn't update unless a delete event occurs).

Similarily, I have also tried creating a select field in the outer scope `"parameters/condition"`. This has the same issue, the selection works but only updates when an item is removed from the list - in this case I would prefer that each new list item created uses the value of the select when it is first created and does not update after (like selecting a recipe to generate a new item, recipe 1 creates a few inputs while recipe 2 creates a few checkboxes for example).

Does anyone know where the issue created for this lives? Was it solved? If so, how do I implement it (documentation)?

**Appendix:**
A similar question - https://stackoverflow.com/questions/49172084/how-to-use-a-when-in-a-list-item-referencing-the-proper-path

**Comment by kumilingus** [comment]
Hi, using relative paths and wildcards will be possible in the next release in May. If you can't wait for this till May, please drop me an email at roman[at]client.io.

**Comment by redwolf-albath** [comment]
Thanks for response! Waiting is fine, glad to hear it's on the way.

**Comment by kumilingus** [comment]
Available in `v3.2.0`.
